example/spec/parallel/s3Access/teaS3CredentialsSpec.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const { URL } = require('url');
+const { STS } = require('aws-sdk');
+
+const { models: { AccessToken } } = require('@cumulus/api');
+const {
+  EarthdataLogin: { getEarthdataAccessToken },
+  distributionApi: { invokeS3CredentialsLambda },
+} = require('@cumulus/integration-tests');
+
+const { setDistributionApiEnvVars } = require('../../helpers/apiUtils');
+const { loadConfig } = require('../../helpers/testUtils');
+
+describe('When accessing s3credentials endpoint', () => {
+  let config;
+
+  beforeAll(async () => {
+    config = await loadConfig();
+    process.env.stackName = config.stackName;
+    process.env.AccessTokensTable = `${config.stackName}-s3-credentials-access-tokens`;
+
+    setDistributionApiEnvVars();
+  });
+
+  describe('an unauthenticated request', () => {
+    it('redirects to Earthdata login for requests on /s3credentials endpoint.', async () => {
+      const response = await invokeS3CredentialsLambda('/s3credentials');
+      const authorizeUrl = new URL(response.headers.location);
+      expect(authorizeUrl.origin).toEqual(process.env.EARTHDATA_BASE_URL);
+      expect(authorizeUrl.searchParams.get('state')).toEqual('/s3credentials');
+      expect(authorizeUrl.pathname).toEqual('/oauth/authorize');
+    });
+  });
+
+  describe('with token associated with an Earthdata Login ID', () => {
+    let accessToken;
+    let accessTokensModel;
+    let accessTokenResponse;
+    let username;
+
+    beforeAll(async () => {
+      accessTokensModel = new AccessToken();
+      username = process.env.EARTHDATA_USERNAME;
+      accessTokenResponse = await getEarthdataAccessToken({
+        redirectUri: process.env.DISTRIBUTION_REDIRECT_ENDPOINT,
+        requestOrigin: process.env.DISTRIBUTION_ENDPOINT,
+        storeAccessToken: false,
+      }).catch((error) => {
+        console.log(error);
+        throw error;
```

Comment:
Will this actually throw and stop running everything else if there is an error? I think the pattern we having been doing as a workaround to this jasmine issue is 
```
if (beforeAllFailed) fail('beforeAll() failed');
``` 

in the `it` block